baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java

```diff
@@ -156,21 +172,45 @@ protected void defaultFileSets() {
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
+        validateMinimumVersion();
         loadMigrations();
-
+        sortMigrationsInGroups(groups);
         defaultFileSets();
 
         BatonExecutionSummary summary = performMigration(groups);
         getLog().info(summary.getSummary());
 
     }
 
+    private void validateMinimumVersion() {
+        Pattern pattern = Pattern.compile(minVersionPattern);
+        Matcher matcher = pattern.matcher(this.minimumVersion);
+        if(!matcher.find()) {
+            this.minimumVersion = "0.0.0";
+            getLog().warn(String.format("Invalid format for minimum version %s, " + 
+                    "setting to 0.0.0. Must be of the format [Number].[Number].[Number]", this.minimumVersion));
+        }
+    }
+
+    protected void sortMigrationsInGroups(List<GroupTarget> groupTargets) {
+        for(GroupTarget groupTarget : groupTargets) {
+            if(groupTarget.getType().equals(VERSIONED)) {
+                Collections.sort(groupTarget.getMigrations(), new Comparator<MigrationTarget>() {
+                    @Override
+                    public int compare(MigrationTarget o1, MigrationTarget o2) {
+                        return o1.getVersion().compareTo(o2.getVersion());
+                    }
+                });
+            }
+        }
+    }
+    
     BatonExecutionSummary performMigration(Collection<GroupTarget> groupTargets) {
         BatonExecutionSummary executionSummary = new BatonExecutionSummary();
         for(GroupTarget groupTarget : groupTargets) {
             GroupSummary groupSummary = new GroupSummary(groupTarget.getGroup());
             for (MigrationTarget migrationTarget : groupTarget.getMigrations()) {
-                if (isActive(migrationTarget)) {
+                if (isActive(migrationTarget) && isMinimumVersion(migrationTarget, groupTarget.getType())) {
```

Comment:
Is it important to notify the user if a migration doesn't meet the minimum version criteria?  i.ie log a warning if it doesn't pass